clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaStoreIngestionService.java

```diff
@@ -397,7 +394,7 @@ public void handleStoreDeleted(Store store) {
         new LandFillObjectPool<>(KafkaMessageEnvelope::new));
 
     aggKafkaConsumerService = new AggKafkaConsumerService(
-        pubSubClientsFactory.getConsumerAdapterFactory(),
+        PubSubClientsFactory.fromVeniceProperties(clusterProperties).getConsumerAdapterFactory(),
```

Comment:
same: Let’s avoid creating factories at individual call sites. Instead, we should create the factory
once per server component and reuse it.